src/trio/_tests/test_highlevel_open_tcp_stream.py

```diff
@@ -370,7 +372,11 @@ async def run_scenario(
     trio.socket.set_custom_socket_factory(scenario)
 
     try:
-        stream = await open_tcp_stream("test.example.com", port, **kwargs)
+        stream = await open_tcp_stream(
+            "test.example.com",
+            port,
+            **kwargs,  # type: ignore[arg-type]
+        )
```

Comment:
[add a comment explaining] why can't open_tcp_stream kwargs be typed?

Reply:
right, because the latter allows passing 2x `<type>`. I suspect the reason for the `kwargs` in the first place is laziness/ease of maintainability, but given we're digging this hard anyway you could probably replace them all. You might want to consider a separate PR though

Reply:
Added comment in c4fccd0

Reply:
rereading my previous comment I realize I probably should've been more explicit: what I was suggesting was to stop using `**kwargs` altogether and simply define the parameters to `run_scenario` to be equivalent to the ones that `open_tcp_stream` expects

Reply:
Fixed in 42be7f3